### DISCUSSION TITLE
Can't install Homebrew on iMac (iMac18,3) Big Sur 11.2.3

### DISCUSSION THREAD

**Discussion by techscholarau** [Homebrew/discussions#1185] [Getting started]
Recently I rebuilt my iMac and I'm trying to install Homebrew but have hit an issue. I have tried the following standard install command in iTerm2 (and Terminal) which look to be using zsh by default (not sure if that is important or not!):

/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

and I get this output:

**curl: (35) error:1400443E:SSL routines:CONNECT_CR_SRVR_HELLO:tlsv1 alert inappropriate fallback**

I also came across the exact same error when trying to install oh-my-zsh, so that would suggest it is either a curl related issue or a machine specific related issue. Given the lack of information available on the search engines, I am guessing it is the later!

I cannot find anything helpful to assist online so I thought I'd reach out to see if anyone could point me in the right direction. The information I could gather for curl (35) error indicate it could be a number of things, of which, certificate and permissions are the ones that interest me. The problem is I have no idea where to start.

Any help would be greatly appreciated.

Thanks in advance.

**Comment by SMillerDev** [comment]
First off, this isn't really an issue in homebrew as it can't even connect to the installer. But I'll try and help you debug your network issues anyway:

Are you on a managed (school/business/country) network? Do you have something (firewall/antivirus/whatever) interfering with your connection? What macOS version are you on?

**Reply by techscholarau** [reply]
Cheers - I agree that it doesn't seem to be a Homebrew issue and I appreciate whatever help I can get.

I am on a home network, and are running Big Sur 11.2.3 I do have a firewall but nothing has changed on it for a while.

The curl error seems to be a handshaking issue but I am just not sure how to start troubleshooting it.

**Reply by SMillerDev** [reply]
Can you open the page in a browser?

And if so, can you send the output of `curl -vvv --head https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh`

**Reply by techscholarau** [reply]
I can open it in a browser and the output of that command is as follows:

curl -vvv --head https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh
*   Trying 185.199.111.133...
* TCP_NODELAY set
* Connected to raw.githubusercontent.com (185.199.111.133) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS alert, unknown (598):
* error:1400443E:SSL routines:CONNECT_CR_SRVR_HELLO:tlsv1 alert inappropriate fallback
* Closing connection 0
curl: (35) error:1400443E:SSL routines:CONNECT_CR_SRVR_HELLO:tlsv1 alert inappropriate fallback

**Comment by gromgit** [comment]
What's the output of:
```
curl --version
sw_vers -productVersion
```
?

**Reply by techscholarau** [reply]
curl --version

curl 7.64.1 (x86_64-apple-darwin20.0) libcurl/7.64.1 (SecureTransport) LibreSSL/2.8.3 zlib/1.2.11 nghttp2/1.41.0
Release-Date: 2019-03-27
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: AsynchDNS GSS-API HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL UnixSockets

sw_vers -productVersion

11.2.3

**Reply by gromgit** [reply]
Your curl and OS are definitely supported, so it's something else. Do you have any security software (antivirus, firewall, etc.) running on your machine? If so, try disabling them one at a time and try `curl -vvv --head https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh` again.

Also, what's the output of:
```
curl -L https://github.com > /dev/null || echo OUCH
curl -L https://google.com > /dev/null || echo OUCH
```
?

**Reply by techscholarau** [reply]
Thanks @SMillerDev and @gromgit - I do feel like a real twat now, but want to document this so others don't make a silly mistake like I did.

I made an assumption that as I had this working previously, and that I hadn't changed any firewall/antivirus configuration or software that I discounted that straight out of the gate. In fact I had updated my antivirus software as I re-installed it from scratch! 

Anyway, I turned **off** my antivirus and I can install Homebrew. I turned the antivirus back **on** and tried to install wget using:

brew install wget

And that failed with:

curl: (35) LibreSSL SSL_connect: SSL_ERROR_ZERO_RETURN in connection to d29vzk4ow07wi7.cloudfront.net:443
Error: Failed to download resource "gettext"
Download failed: https://homebrew.bintray.com/bottles/gettext-0.21.big_sur.bottle.tar.gz

I then turned **off** the antivirus and tried again, and it installed!

Now I'll see if I can work out how I can have my antivirus turned on and allow Homebrew to do it's thing.

Signing off (rather embarrassingly!)

Thanks you again for your time and assistance. It is greatly appreciated.
